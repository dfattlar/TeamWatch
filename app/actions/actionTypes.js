"use strict";

export const START_WATCH = "START_WATCH";
export const STOP_WATCH = "STOP_WATCH";
export const TICK = "TICK";
export const RESET_ALL = "RESET_ALL";
export const RESET_TIME = "RESET_TIME";
export const OPEN_MODAL = "OPEN_MODAL";
export const CLOSE_MODAL = "CLOSE_MODAL";
export const NEW_ATHLETE_INPUT = "NEW_ATHLETE_INPUT";
export const ADD_ATHLETE = "ADD_ATHLETE";
export const ADD_ATHLETE_ERROR = "ADD_ATHLETE_ERROR";
export const ADD_SPLIT = "ADD_SPLIT";
export const MODE_CHANGE = "MODE_CHANGE";
export const RESET_ATHLETE_LIST = "RESET_ATHLETE_LIST";
export const ADD_HISTORY = "ADD_HISTORY";
export const REMOVE_ATHLETE_FROM_WATCH = "REMOVE_ATHLETE_FROM_WATCH";
export const ADD_ATHLETE_TO_WATCH = "ADD_ATHLETE_TO_WATCH";
export const DELETE_ATHLETE = "DELETE_ATHLETE";
export const DELETE_HISTORY = "DELETE_HISTORY";
