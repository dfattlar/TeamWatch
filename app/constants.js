"use strict";

export const RACE = "RACE";
export const RELAY = "RELAY";

export const COLORS = {
  PRIMARY: "#010A1E",
  SECONDARY: "#FF0000",
  NAV_BUTTON: "#fff",
  WATCH_BUTTON: "#fff",
  BUTTON_UNDERLAY: "#f3f3f3",
  BUTTON_START: "#FF0000",
  FONT_LIGHT: "#fff",
  FONT_MEDIUM: "#433C3C",
  FONT_ERRROR: "#FF0000",
  BACKGROUND_LIGHT: "#fff",
  BACKGROUND_CONTAINER: "lightgray",
  BACKGROUND_DARK: "#000",
  ATHLETE1: "#00D3FF",
  ATHLETE2: "#0840F4",
  ATHLETE3: "#073184",
  ATHLETE4: "#010A1E",
  ATHLETE5: "#0CE8D1"
};
