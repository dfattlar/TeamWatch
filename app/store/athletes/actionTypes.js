export const ADD_ATHLETE_TO_WATCH = 'ADD_ATHLETE_TO_WATCH';
export const REMOVE_ATHLETE_FROM_WATCH = 'REMOVE_ATHLETE_FROM_WATCH';
export const REMOVE_ALL_ATHLETES_FROM_WATCH = 'REMOVE_ALL_ATHLETES_FROM_WATCH';
export const NEW_ATHLETE_INPUT_FIRST = 'NEW_ATHLETE_INPUT_FIRST';
export const NEW_ATHLETE_INPUT_LAST = 'NEW_ATHLETE_INPUT_LAST';
export const CREATE_ATHLETE = 'CREATE_ATHLETE';
export const CREATE_ATHLETE_ERROR = 'CREATE_ATHLETE_ERROR';
export const RESET_ATHLETE_LIST = 'RESET_ATHLETE_LIST';
