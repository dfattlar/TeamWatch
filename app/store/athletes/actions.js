import * as types from './actionTypes';

export const addAthleteToWatch = (id, name) => {
  return dispatch => {
    dispatch(addingAthleteToWatch(id, name));
  };
};

export const removeAthleteFromWatch = id => {
  return dispatch => {
    dispatch(removingAthleteFromWatch(id));
  };
};

export const removeAllAthletesFromWatch = () => {
  return dispatch => {
    dispatch(removingAllAthletesFromWatch());
  };
};

export const createAthlete = (first, last) => {
  return dispatch => {
    dispatch(creatingAthlete(first, last));
  };
};

export const createAthleteError = error => {
  return dispatch => {
    dispatch(creatingAthleteError(error));
  };
};

const addingAthleteToWatch = (id, name) => ({
  type: types.ADD_ATHLETE_TO_WATCH,
  payload: {id, name},
});

const removingAthleteFromWatch = id => ({
  type: types.REMOVE_ATHLETE_FROM_WATCH,
  id,
});

const removingAllAthletesFromWatch = () => ({
  type: types.REMOVE_ALL_ATHLETES_FROM_WATCH,
});

const creatingAthlete = (first, last) => ({
  type: types.CREATE_ATHLETE,
  payload: {first, last},
});

const creatingAthleteError = error => ({
  type: types.CREATE_ATHLETE_ERROR,
  error,
});
