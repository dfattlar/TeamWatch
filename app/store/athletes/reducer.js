'use strict';

import * as types from './actionTypes';
import React from 'react';
import * as _ from 'lodash';

const initialState = {
  newAthleteInputFirst: '',
  newAthleteInputLast: '',
  addAthleteError: false,
  athleteStore: [],
};

export default function athletes(state = initialState, action = {}) {
  switch (action.type) {
    case types.CREATE_ATHLETE:
      const {first, last} = action.payload;
      let newAthlete = {
        id: Math.random()
          .toString(12)
          .substring(7),
        name: `${first} ${last}`,
        onWatch: false,
      };
      let arrUpdated = [...state.athletesStore, newAthlete];
      return {
        ...state,
        addAthleteError: false,
        athletesStore: arrUpdated,
        newAthleteInputFirst: '',
        newAthleteInputLast: '',
      };
    case types.ADD_ATHLETE_TO_WATCH:
      const updatedAthleteArr = state.athletesStore.map(function(athlete) {
        if (athlete.id === action.payload.id) {
          return {
            ...athlete,
            onWatch: true,
          };
        }
        return athlete;
      });

      return {
        ...state,
        athleteStore: updatedAthleteArr,
      };
    case types.REMOVE_ATHLETE_FROM_WATCH:
      const removeId = action.id;
      const updatedStore = state.athleteStore.map(athlete => {
        if (athlete.id === removeId) {
          return {
            ...athlete,
            onWatch: false,
          };
        }
        return athlete;
      });

      return {
        ...state,
        athleteStore: updatedStore,
      };

    case types.RESET_ATHLETE_LIST:
      let resetAthleteArr = state.athleteStore.map(function(athlete) {
        return {
          ...athlete,
          onWatch: false,
        };
      });
      return {
        ...state,
        athleteStore: resetAthleteArr,
      };

    // case types.DELETE_ATHLETE:
    //   const deleteId = action.id;
    //   const athleteIndex = state.athleteStore.findIndex(athlete => {
    //     return athlete.id === deleteId;
    //   });

    //   const deleteAthleteArr = [
    //     ...state.athleteStore.slice(0, athleteIndex),
    //     ...state.athleteStore.slice(athleteIndex + 1),
    //   ];

    //   return {
    //     ...state,
    //     athleteStore: deleteAthleteArr,
    //   };
    default:
      return state;
  }
}
