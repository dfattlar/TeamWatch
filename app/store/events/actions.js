import * as types from "./actionTypes";

export const addEvent = payload => {
  return dispatch => {
    dispatch(addingEvent(payload));
  };
};

const addingEvent = payload => ({
  type: types.ADD_EVENT,
  payload
});
