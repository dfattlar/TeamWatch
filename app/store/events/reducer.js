'use strict';

import * as types from './actionTypes';
import React from 'react';

const initialState = {
  eventsStore: [],
};

export default function event(state = initialState, action = {}) {
  switch (action.type) {
    case types.ADD_EVENT:
      const id = Math.random()
        .toString(12)
        .substring(4);
      const payloadWithId = {
        ...action.payload,
        id,
      };
      const updatedEvent = [...state.eventsStore, payloadWithId];
      return {
        ...state,
        eventsStore: updatedEvent,
      };
    case types.DELETE_EVENT:
      const deleteId = action.id;
      const eventIndex = state.eventsStore.findIndex(event => {
        return event.id === deleteId;
      });

      const deleteEventArr = [
        ...state.eventsStore.slice(0, eventIndex),
        ...state.eventsStore.slice(eventIndex + 1),
      ];

      return {
        ...state,
        eventsStore: deleteEventArr,
      };
    default:
      return state;
  }
}
