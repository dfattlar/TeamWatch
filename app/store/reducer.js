import {combineReducers} from 'redux';

import session from './session';
import watch from './watch';
import athletes from './athletes';
import events from './events';

export default combineReducers({
  session,
  watch,
  athletes,
  events,
});
