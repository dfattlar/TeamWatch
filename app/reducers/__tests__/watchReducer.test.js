import reducer from '../watchReducer';
import * as types from '../../actions/actionTypes';
import {RACE, RELAY} from '../../constants';

const initialState = {
  watchRunning: false,
  startTime: null,
  stopTime: null,
  id: 0,
  currentColorId: 0,
  modalVisible: false,
  newAthlete: '',
  athletesArray: [],
  timerMode: RACE,
  lastRelaySplit: null,
  relayFinishTime: null,
  bestTime: 0,
  watchReset: false,
};

describe('watch reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toEqual(initialState);
  });

  it('should handle START_WATCH with no athletes', () => {
    expect(
      reducer(undefined, {
        type: types.START_WATCH,
      }),
    ).toMatchSnapshot();
  });

  it('should handle START_WATCH with athletes (clear total time)', () => {
    expect(
      reducer(
        {
          ...initialState,
          watchRunning: true,
          startTime: 999,
          athletesArray: [
            {name: 'ath1', totalTime: '111'},
            {name: 'ath2', totalTime: '222'},
          ],
          lastRelaySplit: undefined,
        },
        {
          type: types.START_WATCH,
        },
      ),
    ).toMatchSnapshot();
  });

  it('should handle STOP_WATCH', () => {
    const stop = new Date('2019-05-14T11:01:58.135Z').valueOf();
    expect(
      reducer(undefined, {
        type: types.STOP_WATCH,
        stopTime: stop,
      }),
    ).toMatchSnapshot();
  });

  it('should handle STOP_WATCH and set athlete totalTime', () => {
    const stop = new Date('2019-05-14T11:01:58.135Z').valueOf();
    expect(
      reducer(
        {
          ...initialState,
          athletesArray: [
            {name: 'ath1', totalTime: 6, splits: [1, 2, 3]},
            {name: 'ath2', totalTime: 15, splits: [4, 5, 6]},
          ],
          startTime: 100,
          time: 22,
        },
        {
          type: types.STOP_WATCH,
          stopTime: stop,
        },
      ),
    ).toMatchSnapshot();
  });

  it('should handle STOP_WATCH in Relay Mode', () => {
    const stop = new Date('2019-05-14T11:01:58.135Z').valueOf();
    expect(
      reducer(
        {
          ...initialState,
          timerMode: RELAY,
          startTime: 100,
          time: 25,
          athletesArray: [
            {name: 'ath1', totalTime: 6, splits: [1, 2, 3]},
            {name: 'ath2', totalTime: 15, splits: [4, 5, 6]},
            {name: 'ath3', totalTime: '', splits: []},
          ],
        },
        {
          type: types.STOP_WATCH,
          stopTime: stop,
        },
      ),
    ).toMatchSnapshot();
  });

  it('should handle RESET_ALL', () => {
    expect(
      reducer(
        {
          ...initialState,
          watchRunning: true,
          athletesArray: [1, 2, 3],
          id: 10,
          startTime: 10,
          relayFinishTime: 100,
        },
        {type: types.RESET_ALL},
      ),
    ).toMatchSnapshot();
  });

  it('should handle RESET_TIME', () => {
    expect(
      reducer(
        {
          ...initialState,
          watchRunning: true,
          athletesArray: [
            {splits: [1, 2, 3], totalTime: 6},
            {splits: [4, 5, 6], totalTime: 15},
          ],
          startTime: 10,
          relayFinishTime: 10,
        },
        {type: types.RESET_TIME},
      ),
    ).toMatchSnapshot();
  });

  it('should handle OPEN_MODAL', () => {
    expect(reducer(initialState, {type: types.OPEN_MODAL})).toMatchSnapshot();
  });

  it('should handle CLOSE_MODAL', () => {
    expect(
      reducer(
        {
          ...initialState,
          modalVisible: true,
        },
        {type: types.CLOSE_MODAL},
      ),
    ).toMatchSnapshot();
  });

  it('should handle ADD_ATHLETE_TO_WATCH', () => {
    expect(
      reducer(
        {
          ...initialState,
          modalVisible: true,
          addAthleteError: true,
          athletesArray: [
            {
              id: 'id_000',
              name: 'athlete1',
              splits: [],
              colorId: 5,
              totalTime: '',
            },
          ],
          id: 2,
          currentColorId: 5,
        },
        {
          type: types.ADD_ATHLETE_TO_WATCH,
          payload: {id: 'id_123', name: 'athlete2'},
        },
      ),
    ).toMatchSnapshot();
  });

  it('should handle REMOVE_ATHLETE_FROM_WATCH', () => {
    expect(
      reducer(
        {
          ...initialState,
          athletesArray: [{id: 1}, {id: 2}, {id: 3}],
        },
        {type: types.REMOVE_ATHLETE_FROM_WATCH, id: 2},
      ),
    ).toMatchSnapshot();
  });

  it('should handle DELETE_ATHLETE', () => {
    expect(
      reducer(
        {
          ...initialState,
          athletesArray: [{id: 1}, {id: 2}, {id: 3}],
        },
        {type: types.DELETE_ATHLETE, id: 2},
      ),
    ).toMatchSnapshot();
  });

  it('should handle DELETE_ATHLETE - athlete not on watch', () => {
    expect(
      reducer(
        {
          ...initialState,
          athletesArray: [{id: 1}, {id: 2}, {id: 3}],
        },
        {type: types.DELETE_ATHLETE, id: 4},
      ),
    ).toMatchSnapshot();
  });

  it('should handle ADD_SPLIT - No start time', () => {
    expect(
      reducer({initialState}, {type: types.ADD_SPLIT, id: 1, splitTime: 10}),
    ).toMatchSnapshot();
  });

  it('should handle ADD_SPLIT - RACE, First Split', () => {
    expect(
      reducer(
        {
          ...initialState,
          startTime: 1,
          athletesArray: [{id: 'id_123', splits: []}],
        },
        {type: types.ADD_SPLIT, id: 'id_123', splitTime: 10},
      ),
    ).toMatchSnapshot();
  });

  it('should handle ADD_SPLIT - RACE, Multiple Athletes', () => {
    expect(
      reducer(
        {
          ...initialState,
          startTime: 1,
          athletesArray: [
            {id: 'id_123', splits: [1, 2], totalTime: 3},
            {id: 'id_456', splits: [1, 2]},
          ],
          lastRelaySplit: 5,
        },
        {type: types.ADD_SPLIT, id: 'id_456', splitTime: 10},
      ),
    ).toMatchSnapshot();
  });

  it('should handle ADD_SPLIT - RELAY, Second Athlete', () => {
    expect(
      reducer(
        {
          ...initialState,
          startTime: 1,
          athletesArray: [
            {id: 'id_123', splits: [1, 2], totalTime: 3},
            {id: 'id_456', splits: [], totalTime: ''},
          ],
          lastRelaySplit: 2,
          timerMode: RELAY,
        },
        {type: types.ADD_SPLIT, id: 'id_456', splitTime: 10},
      ),
    ).toMatchSnapshot();
  });

  it('should handle MODE_CHANGE', () => {
    expect(
      reducer(
        {
          ...initialState,
          timerMode: RELAY,
        },
        {type: types.MODE_CHANGE, timerMode: RACE},
      ),
    ).toMatchSnapshot();
  });
});
