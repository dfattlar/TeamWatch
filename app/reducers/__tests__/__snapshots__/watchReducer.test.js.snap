// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`watch reducer should handle ADD_ATHLETE_TO_WATCH 1`] = `
Object {
  "addAthleteError": false,
  "athletesArray": Array [
    Object {
      "colorId": 5,
      "id": "id_000",
      "name": "athlete1",
      "splits": Array [],
      "totalTime": "",
    },
    Object {
      "colorId": 0,
      "id": "id_123",
      "name": "athlete2",
      "splits": Array [],
      "totalTime": "",
    },
  ],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 3,
  "lastRelaySplit": null,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": null,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle ADD_SPLIT - No start time 1`] = `
Object {
  "initialState": Object {
    "athletesArray": Array [],
    "bestTime": 0,
    "currentColorId": 0,
    "id": 0,
    "lastRelaySplit": null,
    "modalVisible": false,
    "newAthlete": "",
    "relayFinishTime": null,
    "startTime": null,
    "stopTime": null,
    "timerMode": "RACE",
    "watchReset": false,
    "watchRunning": false,
  },
}
`;

exports[`watch reducer should handle ADD_SPLIT - RACE, First Split 1`] = `
Object {
  "athletesArray": Array [
    Object {
      "id": "id_123",
      "splits": Array [
        9,
      ],
      "totalTime": 9,
    },
  ],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": 10,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": 1,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle ADD_SPLIT - RACE, Multiple Athletes 1`] = `
Object {
  "athletesArray": Array [
    Object {
      "id": "id_123",
      "splits": Array [
        1,
        2,
      ],
      "totalTime": 3,
    },
    Object {
      "id": "id_456",
      "splits": Array [
        1,
        2,
        6,
      ],
      "totalTime": 9,
    },
  ],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": 10,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": 1,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle ADD_SPLIT - RELAY, Second Athlete 1`] = `
Object {
  "athletesArray": Array [
    Object {
      "id": "id_123",
      "splits": Array [
        1,
        2,
      ],
      "totalTime": 3,
    },
    Object {
      "id": "id_456",
      "splits": Array [
        8,
      ],
      "totalTime": 8,
    },
  ],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": 10,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": 1,
  "stopTime": null,
  "timerMode": "RELAY",
  "watchReset": false,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle CLOSE_MODAL 1`] = `
Object {
  "athletesArray": Array [],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": null,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": null,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle DELETE_ATHLETE - athlete not on watch 1`] = `
Object {
  "athletesArray": Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 2,
    },
    Object {
      "id": 3,
    },
  ],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": null,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": null,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle DELETE_ATHLETE 1`] = `
Object {
  "athletesArray": Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 3,
    },
  ],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": null,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": null,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle MODE_CHANGE 1`] = `
Object {
  "athletesArray": Array [],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": null,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": null,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle OPEN_MODAL 1`] = `
Object {
  "addAthleteError": false,
  "athletesArray": Array [],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": null,
  "modalVisible": true,
  "newAthlete": "",
  "newAthleteInput": "",
  "relayFinishTime": null,
  "startTime": null,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle REMOVE_ATHLETE_FROM_WATCH 1`] = `
Object {
  "athletesArray": Array [
    Object {
      "id": 1,
    },
    Object {
      "id": 3,
    },
  ],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": null,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": null,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle RESET_ALL 1`] = `
Object {
  "athletesArray": Array [],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": null,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": null,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": true,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle RESET_TIME 1`] = `
Object {
  "athletesArray": Array [
    Object {
      "splits": Array [],
      "totalTime": "",
    },
    Object {
      "splits": Array [],
      "totalTime": "",
    },
  ],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": null,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": null,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": true,
  "watchRunning": false,
}
`;

exports[`watch reducer should handle START_WATCH with athletes (clear total time) 1`] = `
Object {
  "athletesArray": Array [
    Object {
      "name": "ath1",
      "totalTime": "",
    },
    Object {
      "name": "ath2",
      "totalTime": "",
    },
  ],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": undefined,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": 999,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": true,
}
`;

exports[`watch reducer should handle START_WATCH with no athletes 1`] = `
Object {
  "athletesArray": Array [],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": undefined,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": null,
  "startTime": undefined,
  "stopTime": null,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": true,
}
`;

exports[`watch reducer should handle STOP_WATCH 1`] = `
Object {
  "athletesArray": Array [],
  "bestTime": 0,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": null,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": 0,
  "startTime": null,
  "stopTime": 1557831718135,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": false,
  "watchStop": 1557831718135,
}
`;

exports[`watch reducer should handle STOP_WATCH and set athlete totalTime 1`] = `
Object {
  "athletesArray": Array [
    Object {
      "name": "ath1",
      "splits": Array [
        1,
        2,
        3,
      ],
      "totalTime": 6,
    },
    Object {
      "name": "ath2",
      "splits": Array [
        4,
        5,
        6,
      ],
      "totalTime": 15,
    },
  ],
  "bestTime": 6,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": null,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": 21,
  "startTime": 100,
  "stopTime": 1557831718135,
  "time": 22,
  "timerMode": "RACE",
  "watchReset": false,
  "watchRunning": false,
  "watchStop": 1557831718035,
}
`;

exports[`watch reducer should handle STOP_WATCH in Relay Mode 1`] = `
Object {
  "athletesArray": Array [
    Object {
      "name": "ath1",
      "splits": Array [
        1,
        2,
        3,
      ],
      "totalTime": 6,
    },
    Object {
      "name": "ath2",
      "splits": Array [
        4,
        5,
        6,
      ],
      "totalTime": 15,
    },
    Object {
      "name": "ath3",
      "splits": Array [],
      "totalTime": "",
    },
  ],
  "bestTime": 6,
  "currentColorId": 0,
  "id": 0,
  "lastRelaySplit": null,
  "modalVisible": false,
  "newAthlete": "",
  "relayFinishTime": 21,
  "startTime": 100,
  "stopTime": 1557831718135,
  "time": 25,
  "timerMode": "RELAY",
  "watchReset": false,
  "watchRunning": false,
  "watchStop": 1557831718035,
}
`;
